### PR TITLE
Support profiling $http requests in Angular 1.3+

### PR DESCRIPTION
**Pull request by jadarnel27**
MiniProfiler currently doesn't render profiling for AJAX calls made using the $http service in AngularJS versions starting at 1.3.  This is because the $http service stopped using the `onreadystatechange` event for handling request completion (in favor of the `onload` event) in that version.  Here is the commit for that change: https://github.com/angular/angular.js/commit/40d8da80ce773dd82166428f481e33df51883340

And it's [still like that in the latest version](https://github.com/angular/angular.js/blob/master/src/ng/httpBackend.js#L58) (1.4.3 right now).

The reason that matters is that MiniProfiler [currently depends on that pre-1.3 implementation](https://github.com/MiniProfiler/ui/blob/master/includes.js#L681) (replaces the `onreadystatechange` handler, does MiniProfiler work, calls Angular's handler at the end).

This pull request adds AngularJS 1.3+ support, while maintaining support for earlier versions as well, by trying to use onload if there is no handler for onreadystatechange.

### CODE REVIEW

**Comment by jadarnel27** [comment]
For a tiny bit of context, there are at least a handful of people experiencing this problem based on this Stack Overflow post: [MiniProfiler and AngularJS](http://stackoverflow.com/questions/29922203/miniprofiler-and-angularjs)

**Comment by yellis** [comment]
@SamSaffron this looks ok to me. You ok with it?

**Comment by jadarnel27** [comment]
Sorry to bug you guys.  Any updates on getting this merged in (if @SamSaffron agrees that it's appropriate)?  I was hoping to get this in production for one of my apps somewhat soon.  I can certainly do a custom build with this change for now, but would prefer to be on a legit release =)

**Comment by yellis** [comment]
@SamSaffron  this looks ok to me. unless you have any objections, I will merge tomorrow

File: includes.js [hunk 1]
```diff
@@ -696,6 +696,22 @@ var MiniProfiler = (function () {
                     };
                 }
             }
+			else if (this.onload) {
```
Comment (mjibson) [inline includes.js]: Please fix the whitespace here.

**Comment by jadarnel27** [comment]
@mjibson my mistake, I have corrected that whitespace inconsistencies.

File: includes.js [hunk 2]
```diff
@@ -1111,4 +1127,4 @@ null],["lit",/^[+-]?(?:0x[\da-f]+|(?:(?:\.\d+|\d+(?:\.\d*)?)(?:e[+\-]?\d+)?))/i]
...
...parentNode).nodeType === 1 && !(p = h.getAttribute(d))); if (p !== m) { h = h.parentNode ? h.nodeType === 11 ? 0 : h.getAttribute(d) || 0 : 0; if (!(e = b[m])) { e = f[m]; e = g(e, b[h] || f[h]); e.key = ++i; b[i] = e } c && o(m) } j.removeAttribute(d) } else if (c && (e = a.data(j, "tmplItem"))) { o(e.key); b[e.key] = e; h = a.data(j.parentNode, "tmplItem"); h = h ? h.key : 0 } if (e) { k = e; while (k && k.key != h) { k.nodes.push(j); k = k.parent } delete e._ctnt; delete e._wrap; a.data(j, "tmplItem", e) } function o(a) { a = a + n; e = l[a] = l[a] || g(e, b[e.parent.key + n] || e.parent) } } } function u(a, d, c, b) { if (!a) return l.pop(); l.push({ _: a, tmpl: d, item: this, data: c, options: b }) } function w(d, c, b) { return a.tmpl(a.template(d), c, b, this) } function x(b, d) { var c = b.options || {}; c.wrapped = d; return a.tmpl(a.template(b.tmpl), b.data, c, b.item) } function v(d, c) { var b = this._wrap; return a.map(a(a.isArray(b) ? b.join("") : b).filter(d || "*"), function (a) { return c ? a.innerText || a.textContent : a.outerHTML || s(a) }) } function t() { var b = this.nodes; a.tmpl(null, null, null, this).insertBefore(b[0]); a(b).remove() } })(MiniProfiler.$);
```
Comment (mjibson) [inline includes.js]: Restore newline at EOF

Comment (jadarnel27) [inline includes.js]: That's what I get for trying to do that quickly with the web editor!  I'll fix this up when I get home to my machine this evening.  Thanks for the feedback.

Comment (jadarnel27) [inline includes.js]: @mjibson I corrected that issue last week, but forgot to ping you afterwards (I'm not sure if you automatically get notified of actions in this PR or not).

Comment (mjibson) [inline includes.js]: Err, I think I misled you. It looks like it *doesn't* have a newline at EOL. Whatever the case is, this line should not show up in the diff.

**Comment by akava** [comment]
Any updates on this pull request?
If you are interested in slightly simpler implementation, I've fixed this by replacing if (this.onreadystatechange) with setting empty onreadystatechange function:

    if (this.onreadystatechange == null) {
        this.onreadystatechange = function() {};
    }



Full text:

    XMLHttpRequest.prototype.send = function sendReplacement(data) {
      if (this.onreadystatechange == null) {
          this.onreadystatechange = function() {};
      }

      this._onreadystatechange = this.onreadystatechange;
      this.onreadystatechange = function onReadyStateChangeReplacement() {
        if (this.readyState == 4) {
            var stringIds = this.getResponseHeader('X-MiniProfiler-Ids');
            if (stringIds) {
              var ids = typeof JSON != 'undefined' ? JSON.parse(stringIds) : eval(stringIds);
              fetchResults(ids);
            }
        }

        return this._onreadystatechange.apply(this, arguments);
      }

    return _send.apply(this, arguments);
    }

**Comment by yellis** [comment]
I just merged the changes, and reset the changes to the EOF

**Comment by jadarnel27** [comment]
Thanks, @yellis!

**Comment by Stabzs** [comment]
Will this be included in a new release version soon?  Looks great!  Thanks!

**Comment by yellis** [comment]
This was just included in [nuget 3.2.0.157](https://www.nuget.org/packages/MiniProfiler/3.2.0.157)

**Comment by cottsak** [comment]
Still having problems getting MiniProfiler to work with Angular. Can anyone point me to a good complete reference? 
More https://github.com/SeanKilleen/seankilleen.github.io/issues/199#issuecomment-255909407

**Pull request merged**